lib/Raven/ClientBuilder.php

```diff
@@ -201,6 +206,40 @@ public function removeHttpClientPlugin($className)
         return $this;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function addMiddleware(callable $middleware, $priority = 0)
+    {
+        $this->middlewares[] = [$middleware, $priority];
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function removeMiddleware(callable $middleware)
```

Comment:
How would you effectively remove a middleware? If you created it, you could have the instance with you, but it seems an unlikely case... If you already added it, you shouldn't want to remove it.

But what if you want to remove one of the standard middleware? How would you retrieve the original instance to remove it afterwards?

Reply:
There is a `getMiddlewares()` method that gives you all the registered middlewares with their priorities, so you could loop over the array and do any type of check to find what you want and then remove it. This is particularly useful if you want to remove/replace a middleware. However I noticed that the `Client` class is missing such such method when it should, but as the format in which the middlewares are stored is different from the one used in the client builder I don't know if we should choose to stick with one or if it's fine to use both of them according to the location

Reply:
There is a `getMiddlewares()` method that gives you all the registered middlewares with their priorities, so you could loop over the array and do any type of check to find what you want and then remove it. This is particularly useful if you want to remove/replace a middleware. However I noticed that the `Client` class is missing such method when it should, but as the format in which the middlewares are stored is different from the one used in the client builder I don't know if we should choose to stick with one or if it's fine to use both of them according to the location

Reply:
Do we have a reason for having the different format?

Reply:
Actually (both for the processors and the middlewares) this is done for convenience as there is no need to store temporary data sorted by priority and the format used here is much more easier to handle for operations like removal, find or insertion